app/code/Magento/Inventory/Test/Integration/Stock/GetProductQuantityInStockTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Test\Integration\Stock;
+
+use Magento\Framework\Indexer\IndexerInterface;
+use Magento\Indexer\Model\Indexer;
+use Magento\Inventory\Indexer\Alias;
+use Magento\Inventory\Indexer\IndexNameBuilder;
+use Magento\Inventory\Indexer\IndexStructureInterface;
+use Magento\Inventory\Indexer\StockItemIndexerInterface;
+use Magento\Inventory\Model\GetProductQuantityInStock;
+use Magento\Inventory\Model\ReservationCleanupInterface;
+use Magento\InventoryApi\Api\GetProductQuantityInStockInterface;
+use Magento\InventoryApi\Api\ReservationBuilderInterface;
+use Magento\InventoryApi\Api\ReservationsAppendInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+class GetProductQuantityInStockTest extends TestCase
+{
+    /**
+     * @var IndexerInterface
+     */
+    private $indexer;
+
+    /**
+     * @var ReservationBuilderInterface
+     */
+    private $reservationBuilder;
+
+    /**
+     * @var  ReservationsAppendInterface
+     */
+    private $reservationsAppend;
+
+    /**
+     * @var  ReservationCleanupInterface
+     */
+    private $reservationCleanup;
+
+    /**
+     * @var GetProductQuantityInStock
+     */
+    private $getProductQtyInStockService;
+
+    protected function setUp()
+    {
+        $this->indexer = Bootstrap::getObjectManager()->create(Indexer::class);
+        $this->indexer->load(StockItemIndexerInterface::INDEXER_ID);
+
+        $this->reservationBuilder = Bootstrap::getObjectManager()->get(ReservationBuilderInterface::class);
+        $this->reservationsAppend = Bootstrap::getObjectManager()->get(ReservationsAppendInterface::class);
+
+        $this->getProductQtyInStockService = Bootstrap::getObjectManager()->create(
+            GetProductQuantityInStockInterface::class
+        );
+
+        $this->reservationCleanup = Bootstrap::getObjectManager()->create(ReservationCleanupInterface::class);
+    }
+
+    public function tearDown()
+    {
+        /** @var IndexNameBuilder $indexNameBuilder */
+        $indexNameBuilder = Bootstrap::getObjectManager()->get(IndexNameBuilder::class);
+        /** @var IndexStructureInterface $indexStructure */
+        $indexStructure = Bootstrap::getObjectManager()->get(IndexStructureInterface::class);
+
+        foreach ([1, 2, 3] as $stockId) {
+            $indexName = $indexNameBuilder
+                ->setIndexId(StockItemIndexerInterface::INDEXER_ID)
+                ->addDimension('stock_', $stockId)
+                ->setAlias(Alias::ALIAS_MAIN)
+                ->create();
+            $indexStructure->delete($indexName);
```

Comment:
it's better to make such operation in Rollback Fixture than in tearDown
because you will copy-paste this tear down code among other test classes which use the same fixtures